doc/api/net.md

```diff
@@ -416,109 +571,134 @@ See also: [`socket.setTimeout()`][]
-socket as reported by the operating system. Returns an object with
-three properties, e.g.
-`{ port: 12346, family: 'IPv4', address: '127.0.0.1' }`
+The `socket.address()` method returns the address to which the socket is
+bound, the address family name, and port of the socket as reported by the
+operating system.
+
+The object returned has the following properties:
+
+* `port` {number}
+* `family` {String} Either `'IPv4'` or `'IPv6'`
+* `address` {String} The IPv4 or IPv6 address
 
 ### socket.bufferSize
 <!-- YAML
 added: v0.3.8
 -->
 
-`net.Socket` has the property that `socket.write()` always works. This is to
-help users get up and running quickly. The computer cannot always keep up
-with the amount of data that is written to a socket - the network connection
-simply might be too slow. Node.js will internally queue up the data written to a
-socket and send it out over the wire when it is possible. (Internally it is
-polling on the socket's file descriptor for being writable).
+The read-only `socket.bufferSize` property specifies the current amount of data
+currently pending in the `net.Socket` instances write buffer.
+
+The `net.Socket` class has been designed such that calls to `socket.write()`
+will always work so long as the connection is open. Data written to the socket
+is buffered internally first, then written out to the underlying connection. If
+the data is written to the buffer faster than it can be passed to the connection
+(e.g. when using a slow network connection), the size of the internal buffer
+can grow.
 
-The consequence of this internal buffering is that memory may grow. This
-property shows the number of characters currently buffered to be written.
-(Number of characters is approximately equal to the number of bytes to be
-written, but the buffer may contain strings, and the strings are lazily
-encoded, so the exact number of bytes is not known.)
+The value of `socket.bufferSize` represents the amount of data (in bytes)
+pending in the queue.
 
-Users who experience large or growing `bufferSize` should attempt to
-"throttle" the data flows in their program with [`pause()`][] and [`resume()`][].
+In order to avoid large or growing `socket.bufferSize` numbers, attempts should
+be made to throttle the flow of data written to the socket.
 
 ### socket.bytesRead
 <!-- YAML
 added: v0.5.3
 -->
 
-The amount of received bytes.
+The `socket.bytesRead` property specifies the total number of bytes received.
 
 ### socket.bytesWritten
 <!-- YAML
 added: v0.5.3
 -->
 
-The amount of bytes sent.
+The `socket.bytesWritten` property specifies the total number of bytes written.
 
 ### socket.connect(options[, connectListener])
 <!-- YAML
 added: v0.1.90
 -->
 
-Opens the connection for a given socket.
-
-For TCP sockets, `options` argument should be an object which specifies:
-
-  - `port`: Port the client should connect to (Required).
-
-  - `host`: Host the client should connect to. Defaults to `'localhost'`.
+* `options` {Object}
```

Comment:
all the comments above about linking back to the options-version from the argument-using versions apply to this api set, too